File: projekt/Deck.cpp
#include "stdafx.h"
#include "Deck.h"
#include <algorithm>

Deck::Deck()
{
	for (int i = 0; i<20; i++)
	{
	if (i < 4) deck.push_back(Card("Farmer", 1, 1));
	if (i > 3 && i < 8) deck.push_back(Card("Soldier", 2, 2));
	if (i > 7 && i < 12) deck.push_back(Card("Warrior", 3, 2));
	if (i > 11 && i < 16) deck.push_back(Card("Protector", 2, 3));
	if (i > 15 && i < 19) deck.push_back(Card("General", 3, 3));
	if (i == 19) deck.push_back(Card("King", 4, 4));
	}
	shuffle_deck();
}

Deck::~Deck()
{

}

void Deck::display_deck()
{
	std::cout << "Twoja talia: \n";
	for (unsigned int i = 0; i < deck.size(); i++)
	{
		deck[i].display_card();
		std::cout << "\n";
	}
}

void Deck::shuffle_deck()
{
	std::random_shuffle( deck.begin(), deck.end());
}

Card Deck::remove_top_deck()
{
	Card holder = deck[deck.size() - 1];
	deck.pop_back();
	return holder;
}

void Deck::add_top_deck(Card card)
{
	deck.push_back(card);
}

void Deck::add_to_deck(int position, Card card)
{
	deck.insert(deck.begin() + position, card);
}

Card Deck::remove_target_deck(int position)
{
	Card holder;
	holder = deck[position - 1];
	deck.erase( deck.begin() + position );
	return holder;
}

int Deck::get_size()
{
	return deck.size();
}

void Deck::operator+ (Deck const& second)
{
	for (unsigned int i = 0; i < second.deck.size(); i++)
		deck.push_back(second.deck[i]);
}

Deck Deck::divide_deck(int from)
{
	Deck divided;
	for (unsigned int i = 0; i < deck.size(); i++)
		divided.deck[i] = deck[from + i];
	deck.erase( deck.begin() + from, deck.end() );
	return divided;
}

void Deck::shift_deck(unsigned int to_where)
{
	Deck temp;
	for (unsigned int i = 0; i < to_where; i++)
		{
			temp.deck[i] = deck[i];
		}
	deck.erase(deck.begin(), deck.begin() + to_where);
	for (unsigned int i = 0; i < to_where; i++)
		{
			deck.push_back(temp.deck[i]);
		}
}

File: projekt_java/Deck.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Random;

public class Deck {
	private List<Card> deck = new ArrayList<Card>();
	
	public Deck()
	{
		for (int i = 0; i < 20; i++)
		{
			if(i < 4)
			{
				Card c = new Card("Farmer", 1, 1); 
				deck.add(c);
			}
			if(i > 3 && i < 8)
			{
				Card c = new Card("Soldier", 2, 2);
				deck.add(c);
			}
			if(i > 7 && i < 12)
			{
				Card c = new Card("Warrior", 3, 2);
				deck.add(c);
			}
			if(i > 11 && i < 16)
			{
				Card c = new Card("Protector", 3, 2);
				deck.add(c);
			}
			if(i > 15 && i < 19)
			{
				Card c = new Card("General", 3, 3);
				deck.add(c);
			}
			if (i == 19)
			{
				Card c = new Card("King", 4, 4);
				deck.add(c);
			}
		}
		shuffle_deck();
	}
	
	public void display_deck()
	{
		System.out.println("Twoja talia:");
		for(int i = 0; i < deck.size(); i++)
		{
			deck.get(i).display_card();
		}
	}
	public Card remove_top_deck()
	{
		Card holder = deck.remove(deck.size()-1);
		return holder;
	}
	public void add_top_deck(Card card)
	{
		deck.add(card);
	}
	public int get_size()
	{
		return deck.size();
	}
	public void add_decks(Deck deck1, Deck deck2)
	{
		for(int i = 0; i < deck2.deck.size(); i++)
			deck1.deck.add(deck2.deck.get(i));
	}
	public Deck divide_deck(int from)
	{
		Deck divided = new Deck();
		for(int i = 0; i < deck.size(); i++)
			divided.deck.set(i, deck.get(from + i));
		for(int i = from; i < deck.size(); i++)
			deck.remove(i);
		return divided;
	}
	public void Shift_deck(int to_where)
	{
		Deck temp = new Deck();
		temp.deck.clear();
		for(int i = 0; i < to_where; i++)
			temp.deck.add(deck.get(i));
		for(int i = 0; i < to_where; i++)
			deck.remove(i);
		for(int i = 0; i < to_where; i++)
			deck.add(temp.deck.get(i));
	}
	public Card remove_target(int position)
	{
		Card holder = new Card();
		holder = deck.get(position - 1);
		deck.remove(position);
		return holder;
	}
	public void shuffle_deck()
	{
		long seed = System.nanoTime();
		Collections.shuffle(deck, new Random(seed));
	}
}
